### ISSUE TITLE
Create a new magic variables API

### ISSUE THREAD

**Issue by ryangjchandler** [ryangjchandler/lowland#32]
Alpine provides a similar feature that allows you to create custom `$` prefixed variables that are evaluated at runtime.

The simplest API for this would be:

```js
Lowland.magicProperty('name', function () {

})
```

Plugins would also support a new `magicProperties` key:

```js
Lowland.use({
    magicProperties: {
        name: function () {}
    }
})
```

**Comment by HugoDF** [comment]
Would be cool if the magic property has access to the root element and the element that's using it.

Would mean things like `$dispatch` can be implemented outside of main `Lowland` instance

**Comment by ryangjchandler** [comment]
> Would be cool if the magic property has access to the root element and the element that's using it.
> 
> Would mean things like `$dispatch` can be implemented outside of main `Lowland` instance

Yeah, that's going to be the idea.

**Comment by HugoDF** [comment]
The Alpine list is:
- `$el` (done, easy)
- `$refs` (querySelector magic, just need access to the root element)
- `$event` (done, hard to take out of core probably)
- `$dispatch` (only gotcha is having access to the element on which the directive is)
- `$nextTick` (something needs to track render count and that needs to be passed to magic property)
- `$watch` (hook into the proxy? or have an event triggering thing, in which case it's just sugar for `on`)

**Comment by ryangjchandler** [comment]
I think I might make `$dispatch` a core magic property, since it's something that I use quite often.

**Comment by HugoDF** [comment]
> I think I might make `$dispatch` a core magic property, since it's something that I use quite often.

yeah like `on`, `model`, `bind` right? Bundled but uses the Lowland public API

**Comment by ryangjchandler** [comment]
> > I think I might make `$dispatch` a core magic property, since it's something that I use quite often.
> 
> yeah like `on`, `model`, `bind` right? Bundled but uses the Lowland public API?

Exactly. Would just be part of the `DefaultPlugin`.
